src/main/java/com/powsybl/openloadflow/network/impl/extensions/OverloadManagementSystemImpl.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.network.impl.extensions;
+
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class OverloadManagementSystemImpl implements OverloadManagementSystem {
+
+    private boolean enabled;
+
+    private final String lineIdToMonitor;
+
+    private final double threshold;
+
+    private final String switchIdToOperate;
+
+    private final boolean switchOpen;
+
+    public OverloadManagementSystemImpl(boolean enabled, String lineIdToMonitor, double threshold, String switchIdToOperate, boolean switchOpen) {
+        this.enabled = enabled;
+        this.lineIdToMonitor = Objects.requireNonNull(lineIdToMonitor);
+        this.threshold = threshold;
+        this.switchIdToOperate = Objects.requireNonNull(switchIdToOperate);
+        this.switchOpen = switchOpen;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @Override
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    @Override
+    public String getLineIdToMonitor() {
+        return lineIdToMonitor;
+    }
+
+    @Override
+    public double getThreshold() {
+        return threshold;
+    }
+
+    @Override
+    public String getSwitchIdToOperate() {
```

Comment:
This kind of automation system could use a switch between bus bar sections as remedial action. But most of the time, it is a switch from a line extremity. In that case, I think it is better to model it as a terminal ref (or equipment + side) for performance issue. I think it is the same idea for protection schemes.

Reply:
Sure, but this is something that will be in addition to switch operation and that cannot be implemented right now in OLF because we don't know open a branch at only one side during the simulation